Sources/teaBASE.m

```diff
@@ -122,14 +122,43 @@ - (BOOL)checkForSSHPassphraseICloudKeychainIntegration {
 }
 
 - (BOOL)xcodeCLTInstalled {
-    if ([NSFileManager.defaultManager isExecutableFileAtPath:@"/Library/Developer/CommandLineTools/usr/bin/git"]) {
-        return YES;
+    NSTask *task = [[NSTask alloc] init];
+    [task setLaunchPath:@"/usr/bin/xcode-select"];
+    [task setArguments:@[@"-p"]];
+    
+    NSPipe *nullPipe = [NSPipe pipe];
+    [task setStandardOutput:nullPipe];
+    [task setStandardError:nullPipe];
+    
+    NSError *error = nil;
+    [task launchAndReturnError:&error];
+    
+    if (error) {
+        return NO;
     }
-    //TODO Xcode can be installed anywhere, instead check for the bundle ID with spotlight API or mdfind
-    if ([NSFileManager.defaultManager isExecutableFileAtPath:@"/Applications/Xcode.app"]) {
-        return YES;
+    
+    [task waitUntilExit];
+    return task.terminationStatus == 0;
+}
+
+- (BOOL)xcodeInstalled {
+    NSTask *task = [[NSTask alloc] init];
+    [task setLaunchPath:@"/usr/bin/xcodebuild"];
+    [task setArguments:@[@"-version"]];
```

Comment:
I tried this on a fresh https://tart.run and it pops open the “install xcode command line tools” GUI window. So I don’t think we can use this. The `xcode-select -p` just errors however.

Reply:
I think `mdfind "kMDItemCFBundleIdentifier == 'com.apple.dt.Xcode'"` will work well enoguh

Reply:
Thank you, I will check that again